Lesson2/SecondHomework.playground/Contents.swift

```diff
@@ -0,0 +1,46 @@
+/// An object, which is a list of elements with a type integer, organized according to the LIFO principle.
+class Stack {
+    private var array: [Int] = []
+    
+    func push(element: Int) {
+        array.append(element)
+    }
+    
+    func pop() -> Int? {
+        if array.isEmpty {
+            return nil
+        } else {
+            return array.removeLast()
+        }
+    }
+    
+    func top() -> Int? {
+        return array.last
+    }
+}
+
+/// An object, which is a list of elements with a type integer, organized according to the LIFO principle.
+class StackStatistics: Stack {
+    private var minimalValues: [Int] = []
+    
+    override func push(element: Int) {
+        if minimalValues.isEmpty || element < minimalValues.last! {
```

Comment:
@dersim-davaod, regarding the additional task: do I understand correctly that instead of the array of minimumValues ​​it is necessary to use a different data structure (for example, a linked list)?

Reply:
Anything you want to use, just make sure both time/space-complexity is `O(1)`.

BTW, you can do this using a separate branch.

Reply:
well, thank you